client/rpc/api.ts

```diff
@@ -0,0 +1,52 @@
+import { getAddress } from "viem";
+import { addToStore, getFromStore, StorageKey } from "./storage";
+import { getWalletClient } from "./wallet";
+
+const API_URL = "http://localhost:3210";
+
+const fetchMruInfo = async () => {
+  const response = await fetch(`${API_URL}/info`);
+  const res = await response.json();
+  addToStore(StorageKey.MRU_INFO, res);
+};
+
+const submitAction = async (transition: string, inputs: any) => {
+  const walletClient = await getWalletClient();
```

Comment:
Return error here if no wallet client is found so that `submitAction` can return error and prevent the game from starting.
Right now, game starts even when their is no wallet connected

Reply:
wallet connects as soon as the site is started, so we can stop right there only. No need to handle it on every action tbh